src/main/java/backend/json/JSONStore.java

```diff
@@ -56,6 +57,18 @@
         }
     }
 
+    public void removeStoredRepo(String repoName) {
+        ensureDirectoryExists();
+        try {
+            Path repoJsonFile = Paths.get(RepoStore.directory).resolve(escapeRepoName(repoName));
+            Files.delete(repoJsonFile);
+        } catch (NoSuchFileException e) {
+            logger.error("JSON file does not exist. Repo list is incorrect. ");
+        } catch (IOException e) {
+            logger.error("Unable to delete repo JSON file. ");
+        }
+    }
```

Comment:
Rather than do the I/O in here, you should create a task that does it asynchronously, in case someday we switch to a data store where the operation takes a while. See how `saveRepository` is implemented.